packages/gatsby-theme-newrelic/gatsby/should-update-scroll.js

```diff
@@ -0,0 +1,27 @@
+const shouldUpdateScroll = (
+  { routerProps, getSavedScrollPosition, prevRouterProps },
+  themeOptions
+) => {
+  // Ensures no scrolling ever when search modal is open
+  const params = new URLSearchParams(window?.location?.search);
+  if (params.has('q')) {
+    return false;
+  }
+
+  // If the 'shouldUpdateScroll' theme option exists and is one of the listed routes
+  // OR if a user is changing pages/urls, allow for scrolling to top...
+  if (
+    (themeOptions?.shouldUpdateScroll &&
```

Comment:
this is such a small point, but could we swap the order of these? It is more likely that the pathname will change than the routes matching, so we would want to check that first since it prevents the routes needing to checked